File: backend-node/src/controllers/external.controller.js
const axios = require('axios')
const Player = require('../models/player.model')

// Buscar jugadores en API-Football
exports.searchPlayers = async (req, res) => {
  try {
    const { nombre } = req.query

    const response = await axios.get('https://v3.football.api-sports.io/players', {
    headers: {
        'x-apisports-key': process.env.API_FOOTBALL_KEY
    },
    params: {
        search: nombre,
        league: req.query.league || 140, // 140 = La Liga por defecto
        season: 2023
    }
})

    console.log('Respuesta API:', JSON.stringify(response.data, null, 2))

    const players = response.data.response.map(item => ({
    apiId: item.player.id,
    nombre: item.player.name,
    edad: item.player.age,
    nacionalidad: item.player.nationality,
    imagen: item.player.photo,
    equipo: item.statistics[0]?.team?.name || '',
    liga: item.statistics[0]?.league?.name || '',
    posicion: item.statistics[0]?.games?.position || ''
    }))

    res.json({ result: players, message: 'OK', code: 200 })
  } catch (error) {
    res.status(500).json({ result: null, message: 'Error buscando jugadores', code: 500 })
  }
}

// Importar jugador seleccionado a la base de datos local
exports.importPlayer = async (req, res) => {
  try {
    const { jugadores, geolocalizacion } = req.body

    // jugadores es un array de jugadores seleccionados
    const saved = []
    for (const jugador of jugadores) {
      const exists = await Player.findOne({ apiId: jugador.apiId })
      if (!exists) {
        const player = new Player({ ...jugador, geolocalizacion })
        await player.save()
        saved.push(player)
      }
    }

    res.status(201).json({ 
      result: saved, 
      message: `${saved.length} jugador(es) importado(s)`, 
      code: 201 
    })
  } catch (error) {
    res.status(500).json({ result: null, message: 'Error importando jugadores', code: 500 })
  }
}

File: backend-java/players-service/src/main/java/com/futbolapp/players/service/controller/ExternalController.java
package com.futbolapp.players.service.controller;

import com.futbolapp.players.service.model.Player;
import com.futbolapp.players.service.service.PlayerService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.client.RestTemplate;
import org.springframework.http.HttpHeaders;
import org.springframework.http.HttpEntity;
import org.springframework.http.HttpMethod;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/api/external")
@CrossOrigin(origins = "*")
public class ExternalController {

    @Value("${api-football.key}")
    private String apiKey;

    @Autowired
    private PlayerService playerService;

    @GetMapping("/search")
    public ResponseEntity<?> searchPlayers(
            @RequestParam String nombre,
            @RequestParam(defaultValue = "140") String league,
            @RequestParam(defaultValue = "2023") String season) {
        try {
            RestTemplate restTemplate = new RestTemplate();
            HttpHeaders headers = new HttpHeaders();
            headers.set("x-apisports-key", apiKey);
            HttpEntity<String> entity = new HttpEntity<>(headers);

            String url = "https://v3.football.api-sports.io/players?search=" 
                + nombre + "&league=" + league + "&season=" + season;

            ResponseEntity<Map> response = restTemplate.exchange(
                url, HttpMethod.GET, entity, Map.class);

            List<Map> apiPlayers = (List<Map>) response.getBody().get("response");
            List<Map<String, Object>> players = new ArrayList<>();

            for (Map item : apiPlayers) {
                Map<String, Object> player = new java.util.HashMap<>();
                Map p = (Map) item.get("player");
                List<Map> stats = (List<Map>) item.get("statistics");
                Map stat = stats.get(0);
                Map team = (Map) stat.get("team");
                Map leagueInfo = (Map) stat.get("league");
                Map games = (Map) stat.get("games");

                player.put("apiId", p.get("id"));
                player.put("nombre", p.get("name"));
                player.put("edad", p.get("age"));
                player.put("nacionalidad", p.get("nationality"));
                player.put("imagen", p.get("photo"));
                player.put("equipo", team.get("name"));
                player.put("liga", leagueInfo.get("name"));
                player.put("posicion", games.get("position"));
                players.add(player);
            }

            return ResponseEntity.ok(players);
        } catch (Exception e) {
            return ResponseEntity.status(500).body("Error: " + e.getMessage());
        }
    }

    @PostMapping("/import")
    public ResponseEntity<?> importPlayers(@RequestBody Map<String, Object> body) {
        try {
            List<Map<String, Object>> jugadores = 
                (List<Map<String, Object>>) body.get("jugadores");
            List<Player> saved = new ArrayList<>();

            for (Map<String, Object> j : jugadores) {
                Player player = new Player();
                player.setNombre((String) j.get("nombre"));
                player.setEquipo((String) j.get("equipo"));
                player.setLiga((String) j.get("liga"));
                player.setPosicion((String) j.get("posicion"));
                player.setEdad((Integer) j.get("edad"));
                player.setNacionalidad((String) j.get("nacionalidad"));
                player.setImagen((String) j.get("imagen"));
                if (j.get("apiId") != null) {
                    player.setApiId(((Number) j.get("apiId")).intValue());
                }
                saved.add(playerService.create(player));
            }

            return ResponseEntity.status(201).body(saved);
        } catch (Exception e) {
            return ResponseEntity.status(500).body("Error: " + e.getMessage());
        }
    }
}